extensions/jaeger/runtime/pom.xml

```diff
@@ -32,6 +32,20 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
+            <groupId>io.jaegertracing</groupId>
```

Comment:
Can you measure the amount of additional RSS is present by introducing this dependency when running a native executable?

I'm concerned this approach to supporting Zipkin will add reasonable RSS weight to an application even when they're not using Zipkin compatibility

Reply:
I made what you requested and the results are:
without my changes:

- executable size: 43M
- RSS on startup: 18908
- RSS after first request: 22144

With my changes:
- executable size: 44M
- RSS on startup: 18960
- RSS after first request: 24796

Reply:
Thanks @raulvaldoleiros!

@n1hility, @ebullient what do you think of these numbers? Sufficiently small? I'm inclined to say yes, as adding tracing adds weight anyway, but would appreciate your thoughts

Reply:
I would have add the dependency as optional, explain in the guide that to have zipkin capability you need to add the dependency to your pom.xml, and make a check at runtime that the class is present before enabling Zipkin.

OK, it's a small RSS increase, but still, if we have more and more of those addition we will ends up with a much bigger RSS.

Reply:
I had thought about suggesting that, but wasn't sure if it could be done without causing an issue when it wasn't there.

If it can be done, that would be better.

@raulvaldoleiros can you try?

Reply:
I tried to change the dependency scope to test or provided and it works fine in the quarkus build, but when I'm compiling an application it breaks without the dependency.

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  24.495 s
[INFO] Finished at: 2021-07-09T18:24:08+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (default) on project opentracing-quickstart: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.jaeger.deployment.JaegerProcessor#setupTracer threw an exception: java.lang.NoClassDefFoundError: zipkin2/reporter/Sender

To make the dependency optional the only way is to use java reflection or do you have an example that I can follow?

Reply:
> To make the dependency optional the only way is to use java reflection or do you have an example that I can follow?

Yes, you need to guard the usage of the Zipkin classes by a check that the class exist (Class.forName("") and if an exception occurs, do nothing). 

The only other solution is to create a separate extension but it can be a lot of work so maybe it's not worth it.

Reply:
I changed the code for zipkin be optional, and also I have updated the documentation. Native mode needs extra configuration but it works, I tried to put that configuration in the jaeger extension but I wasn't successful.

Reply:
Is extra configuration for native mode still necessary? If so, what configuration?